util/iter/split.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package iter contains common utilities over iterable types.
+package iter
+
+// Split splits a slice into two slices with the left slice containing all
+// values for which f returns true, and the right containing all values for
+// which f returns false.
+func Split[T any](in []T, f func(T) bool) ([]T, []T) {
+	a := make([]T, 0, len(in))
+	b := make([]T, 0, len(in))
```

Comment:
doesn't seem worth allocating these aggressively. In common cases I can imagine all are false or true.

Or: allocate exactly one slice and fill it in from the front vs back and then reverse the back half at the end.

Do you guarantee ordering? document?